### PR TITLE
Support grafana and the build in dashboard when enable monitor

### LINKED ISSUE #417: support grafana in playground

### PR DESCRIPTION
**Pull request by july2993**
fix #417
Support grafana and the build in dashboard when enable monitor

also, run prometheus with the specified version cause we bind it as the same version of tidb.

```
➜  playground git:(grafana) ✗ tp --monitor
Starting component `playground`: /Users/huangjiahao/code/tiup/components/playground/playground --monitor
Playground Bootstrapping...
Starting component `prometheus`: /Users/huangjiahao/.tiup/components/prometheus/v4.0.0/prometheus/prometheus --config.file=/Users/huangjiahao/.tiup/data/S1UWrQO/prometheus/prometheus.yml --web.external-url=http://127.0.0.1:9090 --web.listen-address=0.0.0.0:9090 --storage.tsdb.path='/Users/huangjiahao/.tiup/data/S1UWrQO/prometheus/data'
start grafana:  tiup grafana --homepath /Users/huangjiahao/.tiup/data/S1UWrQO/grafana --config /Users/huangjiahao/.tiup/data/S1UWrQO/grafana/conf/custom.ini
Starting component `tikv`: /Users/huangjiahao/.tiup/components/tikv/v4.0.0/tikv-server --addr=127.0.0.1:20160 --status-addr=127.0.0.1:20180 --pd=http://127.0.0.1:2379 --config=/Users/huangjiahao/.tiup/data/S1UWrQO/tikv-0/tikv.toml --data-dir=/Users/huangjiahao/.tiup/data/S1UWrQO/tikv-0/data --log-file=/Users/huangjiahao/.tiup/data/S1UWrQO/tikv-0/tikv.log
Starting component `pd`: /Users/huangjiahao/.tiup/components/pd/v4.0.0/pd-server --name=pd-0 --data-dir=/Users/huangjiahao/.tiup/data/S1UWrQO/pd-0/data --peer-urls=http://127.0.0.1:2380 --advertise-peer-urls=http://127.0.0.1:2380 --client-urls=http://127.0.0.1:2379 --advertise-client-urls=http://127.0.0.1:2379 --log-file=/Users/huangjiahao/.tiup/data/S1UWrQO/pd-0/pd.log --initial-cluster=pd-0=http://127.0.0.1:2380
Starting component `tidb`: /Users/huangjiahao/.tiup/components/tidb/v4.0.0/tidb-server -P 4000 --store=tikv --host=127.0.0.1 --status=10080 --path=127.0.0.1:2379 --log-file=/Users/huangjiahao/.tiup/data/S1UWrQO/tidb-0/tidb.log
......
Waiting for tikv 127.0.0.1:20160 ready
CLUSTER START SUCCESSFULLY, Enjoy it ^-^
To connect TiDB: mysql --host 127.0.0.1 --port 4000 -u root
To view the dashboard: http://127.0.0.1:2379/dashboard
To view the Prometheus: http://127.0.0.1:9090
To view the Grafana: http://127.0.0.1:3000

─────────────────────────────────────────────────
```

Screen Shot 2020-06-10 at 2 25 49 PM
Screen Shot 2020-06-10 at 2 26 00 PM

### CODE REVIEW

**Comment by lonng** [comment]
Awesome 👍

File: components/playground/grafana.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
+package main
```
Comment (lonng) [inline components/playground/grafana.go:1]: Please add license header.

File: components/playground/main.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ import (
 	"strings"
 	"time"
 
+	_ "net/http/pprof"
+
```
Comment (lonng) [inline components/playground/main.go:31]: Delete extra blank line.

File: components/playground/playground.go [hunk 3]
```diff
@@ -431,6 +433,58 @@ func (p *Playground) bootCluster(options *bootOptions) error {
 				return
 			}
 		}()
+
+		// set up grafana
+		if err := installIfMissing(p.profile, "grafana", options.version); err != nil {
```
Comment (lonng) [inline components/playground/playground.go:438]: Can we put all `installIfMissing` at the beginning of the start playground and install them concurrently?

File: components/playground/playground.go [hunk 4]
```diff
@@ -395,7 +395,9 @@ func (p *Playground) bootCluster(options *bootOptions) error {
 	}{}
 
 	var monitorCmd *exec.Cmd
+	var grafana *grafana
 	if options.monitor {
```
Comment (lonng) [inline components/playground/playground.go:399]: It's better to run Prometheus and Grafana by default in our new playground.

Comment (july2993) [reply components/playground/playground.go:399]: currently, --monitor control both Prometheus and Grafana, if we want to run it separately we need to add more flag options.
I think we can change --monitor to default true?
I will do this and change the subcommand scale-in/scale-out not hidden when all these things are done if it's ok to make it as true default.

File: components/playground/playground.go (hunk 3 above)
Comment (july2993) [reply components/playground/playground.go:438]: sounds good, can you fire an issue?

Comment (lonng) [reply components/playground/playground.go:438]: https://github.com/pingcap/tiup/issues/491

**Pull request merged**